### PR TITLE
enhance: fix the dev docker yaml and use `go install` to install the gotestsum

### PR DESCRIPTION
**Pull request by SimFG**
/kind improvement

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

File: deployments/docker/dev/docker-compose-apple-silicon.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ services:
   azurite:
     image: mcr.microsoft.com/azure-storage/azurite
     volumes:
-      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/azurite:/opt/azurite
+      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/azurite:/data
```
Comment (LoveEachDay) [inline deployments/docker/dev/docker-compose-apple-silicon.yml:56]: The storage path had been changed for `azurite` ?

Comment (SimFG) [reply deployments/docker/dev/docker-compose-apple-silicon.yml:56]: image
link: https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=docker-hub%2Cblob-storage

Comment (SimFG) [reply deployments/docker/dev/docker-compose-apple-silicon.yml:56]: The dev docker yaml is set incorrectly. If this `/opt/azurite` directory is mapped, the docker will not start.

**Review by LoveEachDay — APPROVED** [review]
/lgtm

**Pull request merged**